### PR TITLE
[HOTFIX] Media version tab

### PR DESCRIPTION
**Pull request by chirimoya**

### CODE REVIEW

File: src/Sulu/Bundle/MediaBundle/Tests/Features/media_upload.feature [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+         When I am editing the media collection "Foobar"
+         Then I expect to see "1" ".item" elements
+         And I double click the data grid item "image1.png"
+         And I expect an overlay to appear
+         And I expect to see "Click or drag and drop new version"
+         And I attach the file "images/image2.png" to the current drop-zone
+         And I expect a success notification to appear
+         And I click the overlay tab "History"
+         And I expect to see "1" ".media-edit-versions-link" elements
+         #FIXME: Why is there only one version
```
Comment (chirimoya) [inline src/Sulu/Bundle/MediaBundle/Tests/Features/media_upload.feature]: @wachterjohannes do you have any idea, why the history tab doesn't get updated correctly? Also the displayed date is not correct.

**Comment by chirimoya** [comment]
@dantleech can you have a look at the behat tests? Are they correct implemented? thx

File: src/Sulu/Bundle/MediaBundle/Tests/Features/media_upload.feature (hunk 1 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/MediaBundle/Tests/Features/media_upload.feature]: no i have no idea when i try it on my machine it seems to work correctly...

Comment (wachterjohannes) [inline src/Sulu/Bundle/MediaBundle/Tests/Features/media_upload.feature]: i think we have to reload by now because the data is not reloaded!

Comment (chirimoya) [inline src/Sulu/Bundle/MediaBundle/Tests/Features/media_upload.feature]: The preview also doesn't get updated.

Comment (wachterjohannes) [inline src/Sulu/Bundle/MediaBundle/Tests/Features/media_upload.feature]: i have found the issue for that! https://github.com/sulu-io/sulu/issues/770

File: src/Sulu/Bundle/MediaBundle/Behat/MediaContext.php [hunk 2]
```diff
@@ -151,6 +153,43 @@ public function iShouldSeeItems($count)
     }
 
     /**
+     * @When I attach the file ":path" to the current drop-zone
+     */
+    public function iAttachTheFileToTheCurrentDropZone($path)
+    {
+
```
Comment (dantleech) [inline src/Sulu/Bundle/MediaBundle/Behat/MediaContext.php]: Remove extra line

File: src/Sulu/Bundle/MediaBundle/Behat/MediaContext.php [hunk 3]
```diff
@@ -151,6 +153,43 @@ public function iShouldSeeItems($count)
...
+                    DIRECTORY_SEPARATOR
+                ) . DIRECTORY_SEPARATOR . ltrim($path, DIRECTORY_SEPARATOR);
+        } else {
+            $fullPath = __DIR__ . DIRECTORY_SEPARATOR . ltrim($path, DIRECTORY_SEPARATOR);
+        }
+
+        if (is_file($fullPath)) {
+            $path = $fullPath;
+        } else {
+            throw new \InvalidArgumentException(
```
Comment (dantleech) [inline src/Sulu/Bundle/MediaBundle/Behat/MediaContext.php]: Inline `sprintf`

````
InvalidArgumentException(sprintf(
````

I believe that is how it is everywhere else ..

File: src/Sulu/Bundle/MediaBundle/Behat/MediaContext.php [hunk 4]
```diff
@@ -151,6 +153,43 @@ public function iShouldSeeItems($count)
...
+                    realpath($this->getMinkParameter('files_path')),
+                    DIRECTORY_SEPARATOR
+                ) . DIRECTORY_SEPARATOR . ltrim($path, DIRECTORY_SEPARATOR);
+        } else {
+            $fullPath = __DIR__ . DIRECTORY_SEPARATOR . ltrim($path, DIRECTORY_SEPARATOR);
+        }
+
+        if (is_file($fullPath)) {
+            $path = $fullPath;
+        } else {
```
Comment (dantleech) [inline src/Sulu/Bundle/MediaBundle/Behat/MediaContext.php]: No need for `else`:

````
if (!is_file($fullPath)) {
    throw new ...
}
````

File: src/Sulu/Bundle/MediaBundle/Tests/Features/media_upload.feature [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+         And the file "image1.png" has been uploaded to the "Foobar" collection
+         When I am editing the media collection "Foobar"
+         Then I expect to see "1" ".item" elements
+         And I double click the data grid item "image1.png"
+         And I expect an overlay to appear
+         And I expect to see "Click or drag and drop new version"
+         And I attach the file "images/image2.png" to the current drop-zone
+         And I expect a success notification to appear
+         And I click the overlay tab "History"
+         And I expect to see "1" ".media-edit-versions-link" elements
```
Comment (dantleech) [inline src/Sulu/Bundle/MediaBundle/Tests/Features/media_upload.feature]: This series of `And` sentances breaks the semantics of the scenario.

`When` should be when you upload a new file, and `Then` should be the success notification.

Try:

````cucumber
Given the media collection...
And the file ...
And I wait to see "1" ".item" elements
And I wait for an overlay to appear
And I wait to see "Click or ...
When I attach the file ...
Then I expect a scucess notification
And ..
And ..
````

i.e. we change `I expect` to `I wait to`. You will need to add the alternative syntax for the `overlay` method in `AdminBundle\Behat\AdminContext`

**Comment by danrot** [comment]
ACK

**Pull request merged**